astropy/io/votable/tests/test_tree.py

```diff
@@ -175,3 +177,316 @@ def test_votable_tag():
+    # Push the VOTable in an IOSTream (emulates a disk saving)
+    buff = io.BytesIO()
+    vtf.to_xml(buff)
+
+    # Read the IOStream (emulates a disk readout)
+    buff.seek(0)
+    vtf2 = parse(buff)
+    assert len(vtf2.resources) == 1
+    for resource in vtf2.resources:
+        assert _squash_xml(mivot_block.content) == _squash_xml(
+            resource.mivot_block.content
+        )
+        assert len(resource.tables) == 1
+
+
+def test_mivot_forbidden_write():
+    """
+    Build a meta resource containing a MIVOT block,
+    build the dummy MIVOT block first.
+    """
+    mivot_block = MivotBlock(
+        """
+    <VODML xmlns="http://www.ivoa.net/xml/mivot" >
+       <REPORT status="KO">Unit test mivot block1</REPORT>
+       <GLOBALS/>
+    </VODML>
+    """
+    )
+    # package the MIVOT block in the resource
+    mivot_resource = Resource()
+    mivot_resource.type = "results"
+
+    with pytest.raises(E26):
+        # A MIVOT block must be with "type=meta"
+        mivot_resource.mivot_block = mivot_block
+
+
+def test_mivot_order(tmp_path):
+    """
+    Build a VOTable with 2 resources containing MivotBlock, parse it, and write it in a file.
+    Then compare it with another file to see if the order of the elements in a resource is respected,
+    in particular the MivotBlock which should be before the tables.
+    """
+    vot = tree
+    mivot_block = MivotBlock(
+        """
+    <VODML xmlns="http://www.ivoa.net/xml/mivot" >
+    </VODML>
+    """
+    )
+    vtf = vot.VOTableFile()
+
+    mivot_resource = Resource()
+    mivot_resource.type = "meta"
+    mivot_resource.mivot_block = mivot_block
+
+    mivot_resource2 = Resource()
+    mivot_resource2.type = "meta"
+    mivot_resource2.mivot_block = mivot_block
+
+    # R1 : 2 mivot_block, 2 tables, 1 description, 1 info, 1 CooSys
+    r1 = vot.Resource()
+    r1.type = "results"
+
+    t1 = vot.Table(vtf)
+    t1.name = "t1"
+    t2 = vot.Table(vtf)
+    t2.name = "t2"
+
+    r1.tables.append(t1)
+    r1.tables.append(t2)
+
+    r1.resources.append(mivot_resource)
+    r1.resources.append(mivot_resource2)
+
+    cs = vot.CooSys(ID="_XYZ", system="ICRS")
+    r1.coordinate_systems.append(cs)
+    i1 = vot.Info(name="test_name", value="test_value")
+    r1.infos.append(i1)
+
+    vtf.resources.append(r1)
+
+    # R2 : 1 resource "results", 1 mivot_block and 1 table
+    r2 = vot.Resource()
+    r2.type = "results"
+
+    r3 = vot.Resource()
+    r3.type = "results"
+
+    t3 = vot.Table(vtf)
+    t3.name = "t3"
+    r2.tables.append(t3)
+    r2.resources.append(mivot_resource)
+    r2.resources.append(r3)
+
+    vtf.resources.append(r2)
+
+    # Push the VOTable in an IOSTream (emulates a disk saving)
+    buff = io.BytesIO()
+    vtf.to_xml(buff)
+
+    # Read the IOStream (emulates a disk readout)
+    buff.seek(0)
+    vtf2 = parse(buff)
+
+    data_path = os.path.dirname(os.path.realpath(__file__))
+    vpath = os.path.join(data_path, "data/test.order.xml")
```

Comment:
I think you can just use our built-in `get_pkg_data_filename()` from `astropy.utils.data`.
```suggestion
    vpath = get_pkg_data_filename("data/test.order.xml")
```